File: llm_toolkit/pydantic_models/message.py
from enum import StrEnum
from dataclasses import dataclass

from pydantic import BaseModel


class Role(StrEnum):
    assistant = 'assistant'
    user = 'user'
    system = 'system'
    archive = 'archive'
    hidden = 'hidden'


class Message(BaseModel):
    thread_uid: str | int
    order: int
    role: Role
    text: str
    archive_for: list[int] | None = None

    def __str__(self):
        if self.role == Role.hidden:
            return f'HIDDEN MESSAGE OF {self.thread_uid} thread'

        return super().__str__()


@dataclass
class SceneArchivingThread:
    background: list[Message]
    messages: list[Message]
    archive: Message | None = None

File: frontend/llm_toolkit_UI/src/models/Message.ts

export enum Role {
    assistant = "assistant",
    user = "user",
    system = "system",
};


export default class Message {
    constructor (
        public threadUID: string,
        public order: number,
        public role: Role,
        public text: string,
        public archiveFor: number[]
    ) {}

    static fromDTO(dto: Record<string, string | number>): Message {
        return new Message(
            dto.thread_uid,
            dto.order,
            dto.role,
            dto.text,
            dto.archive_for
        );
    }

    toDTO(): Record<string, string | number> {
        const msg_dto = { ...this };
        msg_dto['thread_uid'] = this.threadUID;
        msg_dto['archive_for'] = this.archiveFor;

        return msg_dto;
    }
};
